src/Microsoft.ApplicationInsights/Channel/InMemoryTransmitter.cs

```diff
@@ -52,10 +55,25 @@ internal InMemoryTransmitter(TelemetryBuffer buffer)
                     TaskContinuationOptions.OnlyOnFaulted);
         }
 
-        internal Uri EndpointAddress
+        internal Uri InMemoryChannelEndpointAddress { get; set; }
+
+        internal Uri EffectiveEndpointAddress
         {
-            get { return this.endpointAddress; }
-            set { Property.Set(ref this.endpointAddress, value); }
+            get
+            {
+                if (this.InMemoryChannelEndpointAddress != null)
+                {
+                    return this.InMemoryChannelEndpointAddress;
+                }
+                else if (TelemetryConfiguration.Active.GetApplicationInsightsEndpointBaseUri() != null)
+                {
+                    return new Uri(TelemetryConfiguration.Active.GetApplicationInsightsEndpointBaseUri(), TelemetryServiceRelativeEndpoint);
```

Comment:
Minor: This code may get executed with every Send() in certain situations. It does not sounds like an overhead in case of batching, but it may lead to an overhead if items are sent individually. It's OK to leave as is to keep it simple unless you see some easy solution (besides heavy change "subscriber" models) to address this.

Reply:
Yeah, I didn't see a way around this pattern, given that we expect the Config may change during runtime.